subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalInputsIntegrationTest.groovy

```diff
@@ -73,4 +105,45 @@ class IncrementalInputsIntegrationTest extends AbstractIncrementalTasksIntegrati
         executesWithRebuildContext()
     }
 
+    def "cannot query non-incremental file input parameters"() {
+        given:
+        previousExecution()
+
+        when:
+        file("inputs/new-input-file.txt") << "new file"
+        buildFile << """
+            tasks.withType(IncrementalTask).configureEach {
+                queryChangesFor.add("nonIncrementalInput")
```

Comment:
This indirection is a bit hard to follow. Can we maybe pass actual code to execute into the task to do the querying instead?

Reply:
I created a new task instead of trying to cram everything in the one task.